src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -470,7 +470,8 @@ void C2_MacroAssembler::fast_unlock(Register objReg, Register boxReg, Register t
 
   bind(LNotRecursive);
 
-  // Release lock.
+  // Set owner to null.
+  // Release to satisfy the JMM
```

Comment:
This comment doesn't make sense to me.  The JMM is the Java Memory Model, and this just releases the lock.  The Java memory model implies memory ordering and optimizations.  The next comment about fence is more meaningful.  Did someone want this comment?

Reply:
Someone did want this comment in each of the cpu-specific files. I do like to use the same comment in all equal places, so I did what someone wanted. But having said that, CPUs are different. In some you need an explicit release-instruction (like a `membar` of some sort), and in others you don't. I do agree that in a cpu-specific file where it's not needed with an explicit release-instruction, this comment makes no (or less sense). What to do?

Reply:
Whoever asked for it, let's see if this is what they wanted.  I thought the fence() comment below it is what was requested.  I agree the comment should be repeated on all platforms though for consistency.

Reply:
I see where it came from in the aarch64 code, and that code does a stlr() to satisfy the JMM.  It's fine.  Leave it.

Reply:
To get the correct semantics for the JMM the releasing of a monitor has to have release semantics (funny that!). So setting the owner to null should be a release_store. On some platforms we get "release" semantics for free.

The fence is needed for correct operation of our own locking implementation and has nothing to do with JMM.

Reply:
But I concede that the comment is less clear on platforms where the release does not appear in the instruction stream. But I think it important that the need for release semantics be documented.